CenterForOpenScience/osf.io
# Store storage region on users and node-osfstorage addons [PLAT-684]

## Purpose

Instead of storing all info about where to store the files in the variable `WATERBUTLER_CREDENTIALS` inside `addons/osfstorage/settings/local.py`, let's store them in the database!

The schema changes include:
- A new model in `osfstorage/models.py` called `Region` -- a new one of these should be created for each new region that can be selected by users in the future. This would happen via a script, or in the admin! 
- A new `UserSettings` model for `osfstorage` for their default settings
- New fields on `NodeSettings` for configuring region

## Changes

- `addons_osfstorage` migration to add Region, UserSettings, and fields to NodeSettings
- OSF migration
    - create default region
    - create osfstorage `UserSettings` for every user with default region
- Post save signal to add the `osfstorage` addon to the user
- Tests to make sure default region preferences are maintained when making new `NodeSettings` objects
- Tweaks to make sure the new osfstorage usersettings doesn't mess anything else up


## QA Notes



Not a lot QA should be needed, as this is just laying the backend groundwork for frontend changes, and shouldn't actually affect anything yet. 

The one thing to check for is since this is adding a new addon model, it'd be great to check for:
- uploading and downloading files
- merging users
- any messaging that happens when doing something with addons (I found it with merging users for example, now that osfstorage has a usersettings some additional messaging was happening warning about "the osfstorage addon" which should be not shown!)

## Side Effects
Since `osfatorage` has a `UserSettings` model now, whereas before it didn't, anything that checks for a user's `ADDONS_AVAILABLE` now will include osfstorage's `UserSettings`, and that may not be the desired behavior.

## Ticket
https://openscience.atlassian.net/browse/PLAT-684
